test.sh

```diff
@@ -1,80 +1,48 @@
-if [ -z "$CLUSTER_URL" ]; then
-    echo "CLUSTER_URL is empty/unset, launching new cluster."
-    export CCM_AGENTS=6
-    CLUSTER_INFO=$(${REPO_ROOT_DIR}/tools/launch_ccm_cluster.py)
-    echo "Launched cluster: ${CLUSTER_INFO}"
-    # jq emits json strings by default: "value".  Use --raw-output to get value without quotes
-    export CLUSTER_URL=$(echo "${CLUSTER_INFO}" | jq --raw-output .url)
-    export CLUSTER_ID=$(echo "${CLUSTER_INFO}" | jq .id)
-    export CLUSTER_AUTH_TOKEN=$(echo "${CLUSTER_INFO}" | jq --raw-output .auth_token)
-    CLUSTER_CREATED="true"
-else
-    echo "Using provided CLUSTER_URL as cluster: $CLUSTER_URL"
-    CLUSTER_CREATED=""
+#!/bin/bash
+random_id=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 10 | head -n 1)
+set -e -x
+# Build and upload our framework
+FRAMEWORK=$1
+export UNIVERSE_URL_PATH=frameworks/$FRAMEWORK/$FRAMEWORK-universe-url
+frameworks/$FRAMEWORK/./build.sh aws
+if [ ! -f "$UNIVERSE_URL_PATH" ]; then
+    echo "Missing universe URL file: $UNIVERSE_URL_PATH"
+    exit 1
 fi
 
-# A specific framework can be specified to run its tests
-# Otherwise all tests are run in random framework order
-if [ -n "$1" ]; then
-    run_framework_tests $1
-else
-    for framework in $(ls $REPO_ROOT_DIR/frameworks | while IFS= read -r fw; do printf "%05d %s\n" "$RANDOM" "$fw"; done | sort -n | cut -c7-); do
-        echo "Starting shakedown tests for $framework at "`date`
-        run_framework_tests $framework
-    done
-fi
-
-# Tests succeeded. Out of courtesy, trigger a teardown of the cluster if we created it ourselves.
-# Don't wait for the cluster to complete teardown.
-if [ -n "${CLUSTER_CREATED}" ]; then
-    ${REPO_ROOT_DIR}/tools/launch_ccm_cluster.py trigger-stop ${CLUSTER_ID}
-fi
+# Create out test cluster
+cat <<EOF > config.yaml
+launch_config_version: 1
+deployment_name: dcos-ci-test-infinity-$random_id
+template_url: https://s3.amazonaws.com/downloads.mesosphere.io/dcos-enterprise/testing/master/cloudformation/ee.single-master.cloudformation.json
+provider: aws
+aws_region: us-west-2
+key_helper: true
+template_parameters:
+    AdminLocation: 0.0.0.0/0
+    PublicSlaveInstanceCount: 1
+    SlaveInstanceCount: 6
+ssh_user: core
+EOF
+
+dcos-launch create
+dcos-launch wait
+
+# Setup the SSH key for shakedown to use. This is the only way to configure
+# shakedown to use this ssh key without using the shakdedown CLI
+mkdir -p ~/.ssh/
+cat cluster_info.json | jq -r .ssh_private_key > ~/.ssh/id_rsa
+chmod 600 ~/.ssh/id_rsa
+
+# configure the dcos-cli/shakedown-backend
+CLUSTER_URL=http://`dcos-launch describe | jq -r .masters[0].public_ip`
+dcos config set core.dcos_url $CLUSTER_URL
+dcos config set core.ssl_verify false
+tools/./dcos_login.py
+for url in `cat $UNIVERSE_URL_PATH`; do
+    dcos package repo add --index=0 `echo $url | cut -d / -f 5` $url
+done
+
+set +e
+py.test --teamcity -m "sanity" frameworks/$FRAMEWORK/tests
```

Comment:
This is not intended to be a one-size fits all entrypoint. The entrypoint should be pytest. Definitely needs better documentation on how to do setup, but the gist is that more and more of that should live with the tests so that there isnt a opaque/convoluted externalized setup